modules/flowable-bpmn-converter/src/main/java/org/flowable/bpmn/converter/child/MultiInstanceParser.java

```diff
@@ -82,7 +83,26 @@ public void parseChildElement(XMLStreamReader xtr, BaseElement parentElement, Bp
                             		// it is an expression
                                     multiInstanceDef.setInputDataItem((childElement.get(ELEMENT_MULTIINSTANCE_COLLECTION_EXPRESSION).get(0)).getElementText());
                             	}
-                            }
+                				if (childElement.containsKey(ELEMENT_MULTIINSTANCE_COLLECTION_PARSER)) {
+                					ExtensionElement parserElement = childElement.get(ELEMENT_MULTIINSTANCE_COLLECTION_PARSER).get(0);
+                					Map<String, List<ExtensionAttribute>> attributes = parserElement.getAttributes();
+                					String implementation = null;
+                					String implementationType = null;
+
+                					if (attributes.containsKey(ATTRIBUTE_MULTIINSTANCE_COLLECTION_CLASS)) {
+                						implementationType = String.valueOf(ATTRIBUTE_MULTIINSTANCE_COLLECTION_CLASS);
+                						implementation = attributes.get(ATTRIBUTE_MULTIINSTANCE_COLLECTION_CLASS).get(0).getValue();
+                					} else {
+                						implementationType = String.valueOf(ATTRIBUTE_MULTIINSTANCE_COLLECTION_DELEGATEEXPRESSION);
+                						implementation = attributes.get(ATTRIBUTE_MULTIINSTANCE_COLLECTION_DELEGATEEXPRESSION).get(0).getValue();
+                					}
+
+                					AbstractFlowableCollectionParser parser = (AbstractFlowableCollectionParser) Class.forName(implementation).newInstance();
```

Comment:
I only tested using the class implementation type - not sure how to setup and test the delegate expression type.